web/modules/custom/usergreeting/src/GreetingTime.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\usergreeting;
+
+use Drupal\Component\Datetime\TimeInterface;
+
+/**
+ * Provides a greeting service for users dependent on the time of  day.
+ */
+class GreetingTime {
+
+  /**
+   * The user account.
+   *
+   * @var \Drupal\usergreeting\Drupal\Component\Datetime\TimeInterface
+   */
+  protected $timeOutput;
+
+  /**
+   * Construct the Timeinterface object.
+   *
+   * @param \Drupal\Component\Datetime\TimeInterface $timeOutput
+   *   The timestamp output.
+   */
+  public function __construct(TimeInterface $timeOutput) {
+    $this->timeOutput = $timeOutput;
+  }
+
+  /**
+   * {@inheritdoc}
```

Comment:
This class doesn't extend/implement any other classes. Where does it inherit documentation from?